core/services/ocr2/plugins/ocr2keeper/evm21/registry.go

```diff
@@ -926,3 +936,82 @@ func (r *EvmRegistry) getBlockHash(blockNumber *big.Int) (common.Hash, error) {
 
 	return block.Hash(), nil
 }
+
+func (r *EvmRegistry) getTxBlock(txHash common.Hash) (*big.Int, common.Hash, error) {
+	txr, err := r.client.TransactionReceipt(r.ctx, txHash)
+	if err != nil {
+		return nil, common.Hash{}, err
+	}
+
+	return txr.BlockNumber, txr.BlockHash, nil
+}
+
+func (r *EvmRegistry) getCheckResult(p ocr2keepers.UpkeepPayload, reason, state uint8, retryable bool) ocr2keepers.CheckResult {
```

Comment:
nit. This seems like a checkResult without performData, i.e. a result generated due to some error or the upkeep being ineligible. Can we clarify that in the name somehow?

Reply:
done